### PR TITLE
More HTTPS tests

### PR DESCRIPTION
**Pull request by Giotino**
HTTPS tests for `ciphers`, `honorCipherOrder`, `minVersion` https options

#### Checklist

- [x] I have read the documentation.
- [x] I have included a pull request description of my changes.
- [x] I have included some tests.

### CODE REVIEW

File: test/https.ts [hunk 1]
```diff
@@ -434,3 +434,56 @@ test('client certificate PFX', withHttpsServer(), async (t, server, got) => {
...
+		httpsOptions: {
+			ciphers: `${ciphers[1]!}:${ciphers[0]!}`,
+			honorCipherOrder: true
+		}
+	}).json();
+
+	t.is(response.cipher, ciphers[0]);
+});
+
+test('https request with `minVersion` option',	withHttpsServer({maxVersion: 'TLSv1.2'}),	async (t, server, got) => {
```
Comment (sindresorhus) [inline test/https.ts:473]:
```suggestion
test('https request with `minVersion` option', withHttpsServer({maxVersion: 'TLSv1.2'}),	async (t, server, got) => {
```

File: test/https.ts [hunk 2]
```diff
@@ -434,3 +434,56 @@ test('client certificate PFX', withHttpsServer(), async (t, server, got) => {
...
+const ciphers = tls.getCiphers().map(cipher => cipher.toUpperCase());
+
+test('https request with `ciphers` option', withHttpsServer({ciphers: `${ciphers[0]!}:${ciphers[1]!}:${ciphers[2]!}`}), async (t, server, got) => {
+	server.get('/', (request, response) => {
+		response.json({
+			cipher: (request.socket as any).getCipher().name
+		});
+	});
+
+	const response: any = await got({
```
Comment (szmarczak) [inline test/https.ts:447]: Use: `.json<{cipher: type}>()`

File: test/https.ts [hunk 3]
```diff
@@ -434,3 +434,56 @@ test('client certificate PFX', withHttpsServer(), async (t, server, got) => {
...
+});
+
+test('https request with `honorCipherOrder` option', withHttpsServer({ciphers: `${ciphers[0]!}:${ciphers[1]!}`}), async (t, server, got) => {
+	server.get('/', (request, response) => {
+		response.json({
+			cipher: (request.socket as any).getCipher().name
+		});
+	});
+
+	const response: any = await got({
```
Comment (szmarczak) [inline test/https.ts:463]: here

**Review by szmarczak — APPROVED** [review]

**Comment by szmarczak** [comment]
Good job! 💛

**Pull request merged**